packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -31,16 +32,17 @@ export interface PanelChromeProps {
   padding?: PanelPadding;
   title?: string;
   titleItems?: PanelChromeInfoState[];
-  menu?: React.ReactElement;
-  /** dragClass, hoverHeader, loadingState, and states not yet implemented */
+  menu?: ReactElement;
+  /** dragClass, hoverHeader not yet implemented */
   // dragClass?: string;
   hoverHeader?: boolean;
-  // loadingState?: LoadingState;
-  // states?: ReactNode[];
-  /** @deprecated in favor of prop states
+  onStreamingStop?: () => void;
```

Comment:
I don't think we need this, this needs to be more generic (a callback handler on the titleItem )

Reply:
or rather I think we can make stopping query/streaming (same action) can be handled the same way

Reply:
The streaming is not part of the `titleItems`, as streaming is only set if `loadingState === LoadingState.Streaming`. Right now it is a green circle icon and can be only that. If we expect it from titleItems it would be weird for the developers using PanelChrome, imho, to know that titleItems and loadingState are connected only for this case. 
1. We can, however, expect a prop _streamingNode_ of some sort separately that holds the icon and calls a generic callback on click? It will still be semantically connected to the `loadingState` prop again.

2. This might be great:
>And stopping streaming action can be handled the same way as stop a normal query (via the menu)

and we will not have "(click to stop)" on the streaming icon? 

3. What do you mean by this sentence:
>or rather I think we can make stopping query/streaming (same action) can be handled the same way

?

Reply:
Just that we will have to have a way to cancel non steaming queries (via menu) so we can use that for streaming as well , so we don't need this callback

Reply:
Okay, this is great - I'll set it up in the [next PR for the menu](https://github.com/grafana/grafana/pull/60537) itself.